tools/pipelines/templates/build-npm-package.yml

```diff
@@ -407,6 +412,37 @@ extends:
                         inputs:
                           sourceFolder: '${{ parameters.buildDirectory }}/nyc/report'
                           targetFolder: $(Build.ArtifactStagingDirectory)/codeCoverageAnalysis
+                      - task: Bash@3
+                        displayName: Report Code Coverage Comparison
+                        condition: and(succeededOrFailed(), eq('${{ parameters.reportCodeCoverageComparison }}', true), eq(variables['ReportDirExists'], 'true'), eq(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.TargetBranch'], 'main'))
+                        continueOnError: false
+                        env:
+                          ADO_API_TOKEN: $(System.AccessToken)
+                          GITHUB_API_TOKEN: $(githubPublicRepoSecret)
+                          TARGET_BRANCH_NAME: '$(targetBranchName)'
+                          ADO_BUILD_ID: $(Build.BuildId)
+                          GITHUB_PR_NUMBER: $(System.PullRequest.PullRequestNumber)
+                          ADO_CI_BUILD_DEFINITION_ID_BASELINE: 48
+                          ADO_CI_BUILD_DEFINITION_ID_PR: 11
+                        inputs:
+                          targetType: inline
+                          workingDirectory: '${{ parameters.buildDirectory }}'
+                          script: |
+                            set -eux -o pipefail
+                            REPO_NAME=$(echo $BUILD_REPOSITORY_NAME | cut -d'/' -f2)
+                            REPO_OWNER=$(echo $BUILD_REPOSITORY_NAME | cut -d'/' -f1)
+                            export GITHUB_REPOSITORY_NAME=$REPO_NAME
+                            export GITHUB_REPOSITORY_OWNER=$REPO_OWNER
+                            echo "Github Repository Owner: $GITHUB_REPOSITORY_OWNER"
+                            echo "Github Repository Name: $GITHUB_REPOSITORY_NAME"
+                            echo "Github PR number: $GITHUB_PR_NUMBER"
+                            echo "ADO Build Number: $ADO_BUILD_ID"
+                            echo "Target Branch Name: $TARGET_BRANCH_NAME"
+                            echo "ADO CI BUILD_DEFINITION_ID for baseline: $ADO_CI_BUILD_DEFINITION_ID_BASELINE"
+                            echo "ADO CI BUILD_DEFINITION_ID for PR: $ADO_CI_BUILD_DEFINITION_ID_PR"
+                            echo "Running code coverage comparison"
+                            npm run code-coverage-analysis:report
```

Comment:
You could also call flub directly here, which would use the in-repo version of the report:codeCoverage command instead of the installed version. Might be a lot faster for testing and bug fixing. You don't need a build-tools release in that case.

Reply:
That is:

```suggestion
                            flub report codeCoverage
```

Reply:
If we make the change, then I'd say remove the npm script altogether, so people are not misguided into using it and thinking that it should have the same behavior as what happens in the pipeline (it would be using the build-cli/flub installed by the package, not the in-repo version).

Reply:
@alexvy86 You mean if we do this, then remove the codeCovearge:report command from the root package.json?

Reply:
Yeah. It's not the end of the world to have it there, but it won't use the same flub version that the pipeline will be using here (the one installed from npm in package.json, the latest version in the repo in the pipeline). We just need to be aware of that.